x-pack/platform/plugins/private/data_visualizer/public/lite/file_manager/merge_tools.test.ts

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  createMergedMappings,
+  getMappingClashInfo,
+  getFormatClashes,
+  CLASH_TYPE,
+} from './merge_tools';
+import type { FileWrapper, FileAnalysis } from './file_wrapper';
+
+describe('merge_tools', () => {
+  describe('createMergedMappings', () => {
+    it('should return merged mappings when all mappings are the same', () => {
+      const files = [
+        {
+          getMappings: () => ({ properties: { field1: { type: 'text' } } }),
+        } as never as FileWrapper,
+        {
+          getMappings: () => ({ properties: { field1: { type: 'text' } } }),
+        } as never as FileWrapper,
+      ];
+
+      const result = createMergedMappings(files);
+      expect(result.mergedMappings).toEqual({ properties: { field1: { type: 'text' } } });
+      expect(result.mappingClashes).toEqual([]);
+    });
+
+    it('should return merged mappings when all mappings are the keyword and text', () => {
+      const files = [
+        {
+          getMappings: () => ({ properties: { field1: { type: 'text' } } }),
+        } as never as FileWrapper,
+        {
+          getMappings: () => ({ properties: { field1: { type: 'keyword' } } }),
+        } as never as FileWrapper,
+      ];
+
+      const result = createMergedMappings(files);
+      expect(result.mergedMappings).toEqual({ properties: { field1: { type: 'text' } } });
+      expect(result.mappingClashes).toEqual([]);
+    });
+
+    it('should return merged mappings when all mappings are the text and keyword', () => {
```

Comment:
Nit - as above, to make it clearer how this test is different to the one below, what about something like 'should return merged mappings when a text mapping is being merged with a keyword mapping'?

Reply:
Updated in https://github.com/elastic/kibana/pull/209704/commits/1f6f3fa3218f0d29749e724e6e9f0626f0147cbe